src/components/PasswordInput/PasswordInput.tsx

```diff
@@ -0,0 +1,137 @@
+    const inputElementRef = useRef<HTMLInputElement | null>(null);
+    const [isObfuscated, setIsObfuscated] = useState<boolean>(obfuscated);
+
+    const { isFocusVisible, focusProps } = useFocusRing({ within: true, isTextInput: true });
+    const { isFocusVisible: passwordButtonIsFocusVisible, focusProps: passwordButtonFocusProps } = useFocusRing();
+    const { isFocusVisible: clearButtonIsFocusVisible, focusProps: clearButtonFocusProps } = useFocusRing();
+
+    const handleOnChange = useCallback(() => {
+        onChange?.(inputElementRef.current?.value);
+    }, [inputElementRef, onChange]);
+
+    const handleClear = () => {
+        if (inputElementRef.current) {
+            inputElementRef.current.value = '';
+            handleOnChange();
+        }
+    };
+
+    return (
+        <div
+            {...focusProps}
+            className={merge([
+                'tw-flex tw-items-center tw-justify-between tw-h-9 tw-gap-2 tw-px-3 tw-py-3 tw-transition tw-text-s tw-font-sans tw-relative tw-bg-white dark:tw-bg-transparent tw-border tw-rounded tw-line-strong',
+                size ? '' : 'tw-w-full',
+                disabled
+                    ? 'tw-border-line-x-strong'
+                    : 'hover:tw-border-line-x-strong focus-within:tw-border-line-xx-strong focus-within:hover:tw-border-line-xx-strong',
+                isFocusVisible && !passwordButtonIsFocusVisible && !clearButtonIsFocusVisible && FOCUS_STYLE,
+                status ? validationClassMap[status] : '',
+            ])}
+            data-test-id={dataTestId}
+        >
+            <input
+                aria-label="Password Input"
+                autoComplete={autocomplete ? 'on' : 'off'}
+                data-test-id={`${dataTestId}-input`}
+                disabled={disabled}
+                id={useMemoizedId(propId)}
+                onBlur={onBlur}
+                onFocus={onFocus}
+                onKeyDown={onKeyDown}
+                onChange={handleOnChange}
+                onClick={() => inputElementRef.current?.focus()}
+                placeholder={placeholder}
+                readOnly={readOnly}
+                ref={inputElementRef}
+                required={required}
+                size={size}
+                type={isObfuscated ? InputTypes.Password : InputTypes.Text}
+                value={value}
+                className={merge([
+                    'tw-w-full tw-border-none tw-outline-none tw-bg-transparent tw-hide-input-arrows tw-text-sm',
+                    disabled || readOnly
+                        ? 'tw-text-black-40 tw-placeholder-black-30 dark:tw-text-black-30 dark:tw-placeholder-black-40 hover:tw-cursor-not-allowed'
+                        : 'tw-text-black tw-placeholder-black-60 dark:tw-text-white',
+                ])}
+            />
+            <span className="tw-flex tw-justify-between tw-items-center t-max-w-sm">
+                <button
+                    {...passwordButtonFocusProps}
```

Comment:
Any issue with the Icon Button on weak emphasis that you prefer not to use it here?

Reply:
Just the focus state issues stated above in the first comment.